opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+        url.push_str(endpoint);
+        url.push_str("/api/agent/v3/");
+        url.push_str(&self.config.environment);
+        url.push('/');
+        url.push_str(&self.config.account);
+        url.push_str("/MonitoringStorageKeys/?Namespace=");
+        url.push_str(&self.config.namespace);
+        url.push_str("&Region=");
+        url.push_str(&self.config.region);
+        url.push_str("&Identity=");
+        url.push_str(&encoded_identity);
+        url.push_str("&OSType=");
+        url.push_str(get_os_type());
+        url.push_str("&ConfigMajorVersion=");
+        url.push_str(&version_str);
+        url.push_str("&TagId=");
+        url.push_str(&tag_id);
+
+        let req_id = Uuid::new_v4().to_string();
+
+        let response = self
+            .http_client
+            .get(&url)
+            .header(
+                "User-Agent",
+                format!("{}-{}", agent_identity, agent_version),
+            )
+            .header("x-ms-client-request-id", req_id)
+            .header("Accept", "application/json")
+            .send()
+            .await
+            .map_err(GenevaConfigClientError::Http)?;
+
+        let status = response.status();
+        let body = response.text().await?;
+
+        if status.is_success() {
+            let parsed = match serde_json::from_str::<GenevaResponse>(&body) {
+                Ok(response) => response,
+                Err(_) => {
+                    return Err(GenevaConfigClientError::AuthInfoNotFound(
+                        "No primary diag moniker found in storage accounts".to_string(),
+                    ));
+                }
+            };
+
+            for account in &parsed.storage_account_keys {
+                if account.is_primary_moniker && account.account_moniker_name.contains("diag") {
+                    let moniker_info = MonikerInfo {
+                        name: account.account_moniker_name.clone(),
+                        account_group: account.account_group_name.clone(),
+                    };
+
+                    return Ok((parsed.ingestion_gateway_info, moniker_info));
+                }
+            }
+
+            Err(GenevaConfigClientError::MonikerNotFound(
+                "No primary diag moniker found in storage accounts".to_string(),
+            ))
+        } else {
+            Err(GenevaConfigClientError::RequestFailed {
+                status: status.as_u16(),
+                message: body,
+            })
+        }
+    }
+}
+
+#[inline]
+fn get_os_type() -> &'static str {
+    match std::env::consts::OS {
+        "macos" => "Darwin",
+        "windows" => "Windows",
+        "linux" => "Linux",
+        other => other,
+    }
+}
+
+fn extract_endpoint_from_token(token: &str) -> Result<String> {
+    let parts: Vec<&str> = token.split('.').collect();
+    if parts.len() != 3 {
+        return Err(GenevaConfigClientError::Certificate(
```

Comment:
We should use some other error type here not `Certificate`.

Reply:
using `GenevaConfigClientError::JwtTokenError` now.